core/echo_test.go

```diff
@@ -0,0 +1,117 @@
+package core
+
+import (
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func Test_MultiEcho_Bind(t *testing.T) {
+	t.Run("group already bound", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		cfg := NewServerConfig().HTTP.HTTPConfig
+		m := NewMultiEcho(func() EchoServer {
+			return NewMockEchoServer(ctrl)
+		}, cfg)
+		err := m.Bind("", cfg)
+		assert.EqualError(t, err, "http bind group already exists: ")
+	})
+}
+
+func Test_MultiEcho_Routes(t *testing.T) {
+	t.Run("right HTTP methods are mapped", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		s := NewMockEchoServer(ctrl)
+		s.EXPECT().PATCH("patch", nil)
```

Comment:
you may want to split these up since switching 2 calls will still make this test succeed.